controllers/virtualmachine/virtualmachine_controller.go

```diff
@@ -165,6 +171,39 @@ func classBindingToVMMapperFn(ctx *context.ControllerManagerContext, c client.Cl
 	}
 }
 
+// classToVMMapperFn returns a mapper function that can be used to queue reconcile request
+// for the VirtualMachines in response to an event on the VirtualMachineClass resource when
+// WCP_Namespaced_Class_And_Windows_Support is enabled.
+func classToVMMapperFn(ctx *context.ControllerManagerContext, c client.Client) func(o client.Object) []reconcile.Request {
+	// For a given VirtualMachineClass, return reconcile requests
+	// for those VirtualMachines with corresponding VirtualMachinesClasses referenced
+	return func(o client.Object) []reconcile.Request {
+		class := o.(*vmopv1.VirtualMachineClass)
+		logger := ctx.Logger.WithValues("name", class.Name, "namespace", class.Namespace)
+
+		logger.V(4).Info("Reconciling all VMs referencing a VM class because of a VirtualMachineClass watch")
+
+		// Find all vms that match this vmclass
+		vmList := &vmopv1.VirtualMachineList{}
+		if err := c.List(ctx, vmList, client.InNamespace(class.Namespace)); err != nil {
+			logger.Error(err, "Failed to list VirtualMachines for reconciliation due to VirtualMachineClassBinding watch")
+			return nil
+		}
+
+		// Populate reconcile requests for vms matching the classbinding reference
+		var reconcileRequests []reconcile.Request
+		for _, vm := range vmList.Items {
```

Comment:
Hmm.  With this rework, it will be much easier to only watch for Create and Delete events for a VM Class and only queue reconcile for those events.  That might need more discussion since it will be a behavior change (although a desirable one).  Can you either file a story, or start a conversation on this?